src/Subtitles/STS.cpp

```diff
@@ -2862,6 +2860,26 @@ bool CSimpleTextSubtitle::Open(CTextFile* f, int CharSet, CString name) {
 
 #if USE_LIBASS
 
+void CSimpleTextSubtitle::SetSubRenderSettings(SubRendererSettings settings) {
+    bool wasUsingLibass = subRendererSettings.renderUsingLibass;
+    subRendererSettings = settings;
+    if (settings.renderUsingLibass || wasUsingLibass) {
+        ResetASS();
+    }
+}
+
+void CSimpleTextSubtitle::ResetASS() {
+    UnloadASS();

```

Comment:
For embedded subtitles a temporary file is created for parsing the header. When I was working on the code I had an issue where that triggered an unwanted call to UnloadASS(). I moved the call to inside LoadASSFile (after file existence check). So UnloadASS() should in this piece of code be moved to below "m_renderUsingLibass = false;"

Reply:
I added a fix for that, now.  I think it doesn't need to call it unless marked as loaded.

Previously I called that to clean up anything that might be left over.  Did it cause issues when it hadn't been previously loaded, or just a wasted call?

Reply:
It was something like this:
Embedded sub -> data gets put in temp file for parsing -> sets m_path -> later call to SetDefaultStyle called UnloadASS